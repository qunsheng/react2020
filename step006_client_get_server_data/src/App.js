import React from "react";
import HelloApp1 from './HelloApp1';
import HelloApp2 from './HelloApp2';

class App extends React.Component {
  
  constructor(props) {
      super(props);
      this.state = {
          helloDefault: true
      }
  }

  toggleDefault = () => {
      // if setState parameter is function, the prevSate obj will passed in
      this.setState( prevState => {
          // return new state
          return { helloDefault: !prevState.helloDefault}
      })
  }

  render() {
    const { helloDefault } = this.state;
    return (
      <div>
        <div>
          <button onClick = {this.toggleDefault}> Swith Hello App</button>
        </div>
        {helloDefault? <div>
            <p> From Hello App 1</p>
            <HelloApp1 />
          </div>: <div>
            <p> From Hello App 2</p>
            <HelloApp2 />
          </div>
        }
      </div>
    );
  }
}

export default App;
