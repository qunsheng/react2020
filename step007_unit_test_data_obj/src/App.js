import React from "react";
import AppData from "./AppData";

class App extends React.Component {
  constructor(props) {
    super();
    this.appDataObj = new AppData();
    this.state = { data: { name: "unknown", action: "unknown" } };
  }
  render() {
    return (
      <div>
        <p>name: {this.state.data.name || ""} </p>
        <p>action: {this.state.data.action || ""} </p>
      </div>
    );
  }
  componentDidMount() {
    this.appDataObj
      .dataQuery("hello.json")
      .then(data => this.setState({ data }));
  }
}

export default App;
