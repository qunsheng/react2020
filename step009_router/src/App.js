import React from "react";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";
import UiLink from "@material-ui/core/Link";
import ProTip from "./ProTip";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";

function Copyright() {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {"Copyright Â© "}
      <UiLink color="inherit" href="https://material-ui.com/">
        Your Website
      </UiLink>{" "}
      {new Date().getFullYear()}
      {"."}
    </Typography>
  );
}

export default function App() {
  return (
    <div>
      <Container maxWidth="sm">
        <Box my={4}>
          <Typography variant="h4" component="h1" gutterBottom>
            React App example
          </Typography>
          <ProTip />
          <Copyright />
        </Box>
        <Router>
          <div>
            <nav>
              <ul>
                <li>
                  <Link to="/">Home</Link>
                </li>
                <li>
                  <Link to="/about">About</Link>
                </li>
                <li>
                  <Link to="/users">Users</Link>
                </li>
              </ul>
            </nav>

            {/* A <Switch> looks through its children <Route>s and
                renders the first one that matches the current URL. */}
            <Switch>
              <Route path="/about">
                <About />
              </Route>
              <Route path="/users">
                <Users />
              </Route>
              <Route path="/">
                <Home />
              </Route>
            </Switch>
          </div>
        </Router>
      </Container>
    </div>
  );
}

function Home() {
  return <h2>Home</h2>;
}

function About() {
  return <h2>About</h2>;
}

function Users() {
  return <h2>Users</h2>;
}
