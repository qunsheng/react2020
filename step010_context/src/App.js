import React from "react";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";

import {ThemeContext, themes} from './theme-context';
import ThemedButtonComponent1 from './themed-button-component-1';
import ThemedButtonComponent2 from './themed-button-component-2';
import ThemedButtonConsumer from './themed-button-consumer';
import ThemedButtonHook from './themed-button-hook';

// An intermediate component that uses the ThemedButton
function Toolbar(props) {
  return (
    <ThemedButtonComponent1 onClick={props.changeTheme}>
      Click to Change Theme 
    </ThemedButtonComponent1>
  );
}

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      theme: themes.light,
    };

    this.toggleTheme = () => {
      this.setState(state => ({
        theme:
          state.theme === themes.dark
            ? themes.light
            : themes.dark,
      }));
    };
  }

  render() {
    return <div><Container maxWidth="sm">
        <Box my={4}>
          <Typography variant="h4" component="h1" gutterBottom>
            React App example
          </Typography>
        </Box>

        <Box my={4}>
          <Toolbar changeTheme={this.toggleTheme} />
        </Box>
        {/* Provider component accepts a value prop that allows consuming components to subscribe to context changes.*/}
        <ThemeContext.Provider value={this.state.theme}>
          <Box my={4}>
            <ThemedButtonComponent1 > Component Button 1 </ThemedButtonComponent1>
          </Box>
          <Box my={4}>
            <ThemedButtonComponent2 > Component Button 2 </ThemedButtonComponent2>
          </Box>
          <Box my={4}>
            <ThemedButtonConsumer > Consumer Button  </ThemedButtonConsumer>
          </Box>
          <Box my={4}>
            <ThemedButtonHook > Hook Button </ThemedButtonHook>
          </Box>
        </ThemeContext.Provider>
                  
      
      </Container>
    </div>
  }
}

export default App;


