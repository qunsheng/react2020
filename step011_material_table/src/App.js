import React, { forwardRef} from "react";

import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";

import MaterialTable from "material-table";

import AddBox from '@material-ui/icons/AddBox';
import ArrowDownward from '@material-ui/icons/ArrowDownward';
import Check from '@material-ui/icons/Check';
import ChevronLeft from '@material-ui/icons/ChevronLeft';
import ChevronRight from '@material-ui/icons/ChevronRight';
import Clear from '@material-ui/icons/Clear';
import DeleteOutline from '@material-ui/icons/DeleteOutline';
import Edit from '@material-ui/icons/Edit';
import FilterList from '@material-ui/icons/FilterList';
import FirstPage from '@material-ui/icons/FirstPage';
import LastPage from '@material-ui/icons/LastPage';
import Remove from '@material-ui/icons/Remove';
import SaveAlt from '@material-ui/icons/SaveAlt';
import Search from '@material-ui/icons/Search';
import ViewColumn from '@material-ui/icons/ViewColumn';

const tableIcons = {
    Add: forwardRef((props, ref) => <AddBox {...props} ref={ref} />),
    Check: forwardRef((props, ref) => <Check {...props} ref={ref} />),
    Clear: forwardRef((props, ref) => <Clear {...props} ref={ref} />),
    Delete: forwardRef((props, ref) => <DeleteOutline {...props} ref={ref} />),
    DetailPanel: forwardRef((props, ref) => <ChevronRight {...props} ref={ref} />),
    Edit: forwardRef((props, ref) => <Edit {...props} ref={ref} />),
    Export: forwardRef((props, ref) => <SaveAlt {...props} ref={ref} />),
    Filter: forwardRef((props, ref) => <FilterList {...props} ref={ref} />),
    FirstPage: forwardRef((props, ref) => <FirstPage {...props} ref={ref} />),
    LastPage: forwardRef((props, ref) => <LastPage {...props} ref={ref} />),
    NextPage: forwardRef((props, ref) => <ChevronRight {...props} ref={ref} />),
    PreviousPage: forwardRef((props, ref) => <ChevronLeft {...props} ref={ref} />),
    ResetSearch: forwardRef((props, ref) => <Clear {...props} ref={ref} />),
    Search: forwardRef((props, ref) => <Search {...props} ref={ref} />),
    SortArrow: forwardRef((props, ref) => <ArrowDownward {...props} ref={ref} />),
    ThirdStateCheck: forwardRef((props, ref) => <Remove {...props} ref={ref} />),
    ViewColumn: forwardRef((props, ref) => <ViewColumn {...props} ref={ref} />)
};

export default function App() {
  return (
    <Container maxWidth="lg">
        <Box my={4}>
          <Typography variant="h4" component="h1" gutterBottom>
            React App example
          </Typography>
        </Box>
        <MaterialTable
          title="Basic Tree Data Preview"
          icons={tableIcons}
          data={[
            {
              id: 1,
              name: 'a',
              surname: 'Baran',
              birthYear: 1987,
              birthCity: 63,
              sex: 'Male',
              type: 'adult',
            },
            {
              id: 2,
              name: 'b',
              surname: 'Baran',
              birthYear: 1987,
              birthCity: 34,
              sex: 'Female',
              type: 'adult',
              parentId: 1,
            },
            {
              id: 3,
              name: 'c',
              surname: 'Baran',
              birthYear: 1987,
              birthCity: 34,
              sex: 'Female',
              type: 'child',
              parentId: 1,
            },
            {
              id: 4,
              name: 'd',
              surname: 'Baran',
              birthYear: 1987,
              birthCity: 34,
              sex: 'Female',
              type: 'child',
              parentId: 3,
            },
            {
              id: 5,
              name: 'e',
              surname: 'Baran',
              birthYear: 1987,
              birthCity: 34,
              sex: 'Female',
              type: 'child',
            },
            {
              id: 6,
              name: 'f',
              surname: 'Baran',
              birthYear: 1987,
              birthCity: 34,
              sex: 'Female',
              type: 'child',
              parentId: 5,
            },
          ]}
          columns={[
            { title: 'Adı', field: 'name' },
            { title: 'Soyadı', field: 'surname' },
            { title: 'Cinsiyet', field: 'sex' },
            { title: 'Tipi', field: 'type', removable: false },
            { title: 'Doğum Yılı', field: 'birthYear', type: 'numeric' },
            {
              title: 'Doğum Yeri',
              field: 'birthCity',
              lookup: { 34: 'İstanbul', 63: 'Şanlıurfa' },
            },
          ]}
          parentChildData={(row, rows) => rows.find(a => a.id === row.parentId)}
          options={{
            selection: true,
          }}
        />
    </Container>
  );
}
