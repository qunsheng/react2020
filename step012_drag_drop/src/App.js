import React from "react";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";
import AppDragDropDemo from './AppDragDropDemo';

export default function App() {
  return (
    <div>
      <Container maxWidth="lg">
        <Box my={4}>
          <Typography variant="h4" component="h1" gutterBottom>
            React App example
          </Typography>
        </Box>

        <AppDragDropDemo />

      </Container>
    </div>
  );
}

