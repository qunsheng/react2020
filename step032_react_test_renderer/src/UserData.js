
import axios from "axios";

class UserData {
    readData(page) {
        let url = `https://randomuser.me/api?page=${page}`;
        return axios.get(url).then(r => r.data); // return data field of response object
    }
}

export default UserData;