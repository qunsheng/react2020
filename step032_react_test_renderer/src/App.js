import React, {useState, useEffect, useRef} from "react";
import UserData from './UserData';

// users component
function Users (props) {
  return props.users.map( user => {
    let userInfo = `${user.name.title} ${user.name.first} ${user.name.last}`
    return <div key={user.email}>
      <p><img src={user.picture.medium} alt=""></img> {userInfo}</p>
    </div>
    })
}

function App (props) {
  const [users, setUsers] = useState([]);
  const [page, setPage] = useState(1);
  const getUsersForPage = useRef(() => {});

  // test can pass resolved data through props.dataObj
  const dataObj = props.dataObj || new UserData();

  getUsersForPage.current =  () => {
    dataObj.readData(page).then( data => {
      if(data && data.results && data.results.length) {
        setUsers([...users, ...data.results]);
      }
    }).catch( (err) => {
      console.log("api err: ", err)
    });
    setPage(page+1);
  }

  useEffect( () => {
    getUsersForPage.current();
  }, [] );

  return (
    <div>
      <p></p>
      {users.length === 0 ? <span data-testid="loading">Loading data...</span> :
      <div>
        <h1 data-testid="resolved"> User List </h1>
        <button onClick = {getUsersForPage.current}> Append next user page</button>
        < Users users={users} />
      </div>}
    </div>
  )
}

export default App;