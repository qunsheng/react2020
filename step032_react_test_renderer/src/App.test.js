import React from "react";
import renderer from 'react-test-renderer';
import App from './App';
import UserData from './UserData';

function defer() {
    let res, rej;
    let promise = new Promise((resolve, reject) => {
        res = resolve;
        rej = reject;
    });
    promise.resolve = res;
    promise.reject = rej;
    return promise;
}
it('should render app with loading message', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render a single user', async () => {
    const dataObj = new UserData();
    const deferredData = defer();
    let component; 
    deferredData.resolve({ 
            results: [
                {
                    cell: "0441-038-518",
                    email: "ethan.simmons@example.com",
                    gender: "male",
                    id: {name: 'TFN', value: '166627353'},
                    location: {},
                    name: {title: 'Mr', first: 'Ethan', last: 'Simmons'},
                    phone: "08-5807-4652",
                    picture: {
                        large: 'https://randomuser.me/api/portraits/men/3.jpg', 
                        medium: 'https://randomuser.me/api/portraits/med/men/3.jpg', 
                        thumbnail: 'https://randomuser.me/api/portraits/thumb/men/3.jpg'
                    }
                }
            ] 
    });
    dataObj.readData = () => deferredData;
    await renderer.act ( async () => {
        component = renderer.create(<App dataObj={dataObj} />)
    })
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
