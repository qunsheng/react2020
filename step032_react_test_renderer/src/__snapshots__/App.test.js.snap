// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a single user 1`] = `
<div>
  <p />
  <div>
    <h1
      data-testid="resolved"
    >
       User List 
    </h1>
    <button
      onClick={[Function]}
    >
       Append next user page
    </button>
    <div>
      <p>
        <img
          alt=""
          src="https://randomuser.me/api/portraits/med/men/3.jpg"
        />
         
        Mr Ethan Simmons
      </p>
    </div>
  </div>
</div>
`;

exports[`should render app with loading message 1`] = `
<div>
  <p />
  <span
    data-testid="loading"
  >
    Loading data...
  </span>
</div>
`;
